File: oct/6.py
w,h,n,m = list(map(int,input().split()))
a=list(map(int,input().split()))
b=list(map(int,input().split()))
x,y,sqr = set(),set(),set()

for i in a:
    for j in a:
        if i is not j:
            x.add(abs(i-j))

for i in b:
    for j in b:
        if i is not j:
            y.add(abs(i-j))

for side in x:
    if side in y:
        sqr.add(side)

ans = len(sq)

for i in range(h+1):
    curr = len(sqr)
    for _y in b:
        side = abs(_y-i)
        if side is 0 or side in y:
            continue
        elif side not in y and side in x:
            curr+=1
    
    ans = max(ans,curr)

File: oct/6.cpp
#include <bits/stdc++.h>
using namespace std;
#pragma GCC optimize("Ofast")
#pragma GCC target("avx,avx2,fma")
#pragma GCC optimization ("unroll-loops")
int w,h,n,m;

int main(){
    scanf("%d%d%d%d",&w,&h,&n,&m);
    vector<int> a(n);
    vector<int> b(m);

    for(int i=0;i<n;i++) 
        scanf("%d",&a[i]);
    for(int i=0;i<m;i++) 
        scanf("%d",&b[i]);

    unordered_set<int> x_side;
    unordered_set<int> y_side;
    for(int i = 0; i <n-1;i++){
        for(int j = i+1; j <n;j++){
            x_side.insert(abs(a[i]-a[j]));
        }
    }
    for(int i = 0; i <m-1;i++){
        for(int j = i+1; j <m;j++)
            y_side.insert(abs(b[i]-b[j]));
    }
    unordered_set<int> sq;
    for(auto x:x_side){
        if(y_side.count(x))
            sq.insert(x);
    }
    int ans = sq.size();
    for(int i=0;i<=h;i++){
        int curr = sq.size();
        for(auto y:b){
            int side = abs(y-i);
            if(!side or y_side.count(side)){
                continue;
            }else{
                if(x_side.count(side))
                    curr++;
            }
        }
        ans = max(ans,curr);
    }
    printf("%d\n",ans);


    return 0;
}